chalice/compat.py

```diff
@@ -1,8 +1,112 @@
 import socket
 import six
+import os
 
 from six import StringIO
 
+if os.name == 'nt':
+    # windows
+    # On windows running python in a subprocess with no environment variables
+    # will cause sevral issues. In order for our subprocess to run normally we
+    # manually copy the relevant environment variables from the parent process.
+    subprocess_python_base_environ = {
+        # http://bugs.python.org/issue8557
+        'PATH': os.environ['PATH']
+    }
+    # http://bugs.python.org/issue20614
+    if 'SYSTEMROOT' in os.environ:
+        subprocess_python_base_environ['SYSTEMROOT'] = os.environ['SYSTEMROOT']
+
+    # This is the acutal patch used on windows to prevent distutils from
+    # compiling C extensions. The msvc compiler base class has its compile
+    # method overridden to raise a CompileError. This can be caught by
+    # setup.py code which can then fallback to making a purepython
+    # package if possible.
+    # We need mypy to ignore these since they are never actually called from
+    # within ourprocess they do not need to be a part of our typechecking
+    # pass.
+    def prevent_msvc_compiling_patch():  # type: ignore
+        import distutils
+        import distutils._msvccompiler
+        import distutils.msvc9compiler
+        import distutils.msvccompiler
+
+        from distutils.errors import CompileError
+
+        def raise_compile_error(*args, **kwargs):  # type: ignore
+            raise CompileError('Chalice blocked C extension compiling.')
+        distutils._msvccompiler.MSVCCompiler.compile = raise_compile_error
+        distutils.msvc9compiler.MSVCCompiler.compile = raise_compile_error
+        distutils.msvccompiler.MSVCCompiler.compile = raise_compile_error
+
+    # This is the setuptools shim used to execute setup.py by pip.
+    # Lines 2 and 3 have been added to call the above function
+    # `prevent_msvc_compiling_patch` and extra escapes have been added on line
+    # 5 because it is passed through another layer of string parsing before it
+    # is executed.
+    _SETUPTOOLS_SHIM = (
```

Comment:
The escaping makes this hard to read.  Could you just use `r"""` to avoid all the double escaping?